readme.txt
=== Postie ===
Contributors: WayneAllen
Donate link: https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=HPK99BJ88V4C2
Author URI: http://allens-home.com/
Plugin URI: http://PostiePlugin.com/
Tags: e-mail, email, post-by-email
Requires PHP: 5.3
Requires at least: 4.0
Tested up to: 4.9
Stable tag: 1.9.24
License: GPLv2 or later
License URI: http://www.gnu.org/licenses/gpl-2.0.html

Postie allows you to create posts via email, including many advanced features not found in WordPress's default Post by Email feature.

== Description ==
Postie offers many advanced features for creating posts by email, including the ability to assign categories by name, included pictures and videos, and automatically strip off signatures. 
Postie supports both IMAP and POP including SSL/TLS.
There is also an extensive set of filters/actions for developers to extend Postie's functionality.
For usage notes, see the [other notes](other_notes) page.

More info at http://PostiePlugin.com/

= Features =
* Supports IMAP or POP3 servers
* SSL and TLS supported
* Control who gets to post via email
* Set defaults for category, status, post format, post type and tags.
* Set title, category, status, post format, post type, date, comment control and tags in email to override defaults.
* Specify post excerpt (including excerpt only images).
* Use plain text or HTML version of email.
* Remove headers and footers from email (useful for posting from a mailing list).
* Optionally send emails on post success/failure.
* Control the types of attachments that are allowed by file name (wildcards allowed) and MIME type.
* Optionally make the first image the featured image.
* Gallery support.
* Control image placement with plain text email.
* Templates for images so they look the way you want.
* Templates for videos.
* Templates for audio files.
* Templates for other attachments.
* Email replies become comments.

= Developers =
* Several filter hooks available for custom processing of emails.
* More developer info at <a href="http://postieplugin.com/extending/">http://postieplugin.com/extending/</a>

== Screenshots ==

1. Postie server options
2. Postie user options
3. Postie message options
4. More message options
5. Even more message options
6. Image options
7. Video and Audio options
8. Attachment options

== Installation ==
* Install Postie either via the WordPress.org plugin directory, or by uploading the files to your server.
* Activate Postie through the Plugins menu in WordPress.
* Configure the plugin by going to the Postie menu that appears in your admin menu.
* Make sure you enter the mailserver information correctly, including the type of connection and the port number. 
* More information can be found at <a href="http://postieplugin.com/">http://postieplugin.com/</a>

== Usage ==

Please visit our site at <a href="http://postieplugin.com/">http://postieplugin.com/</a>

== Frequently Asked Questions ==

Please visit our FAQ page at <a href="http://postieplugin.com/faq/">http://postieplugin.com/faq/</a>
== Upgrade Notice ==

= 1.8.23 =
Postie now respects the blog timezone, this may require you to change the "Postie Time Correction" setting.

= 1.8.0 =
The php-imap library has been replaced with logic based on Flourish fMailbox et al, there are some differences in the structure of the mail header array. This affects the 
postie_filter_email3 and postie_post_before filters.
See http://flourishlib.com/docs/fMailbox

= 1.6.0 =
Remote cron jobs need to update the URL used to kick off a manual email check. The new URL is http://<mysite>/?postie=get-mail
Accessing http://<mysite>/wp-content/plugins/postie/get_mail.php will now receive a 403 error and a message stating what the new URL should be.
The Postie menu is now at the main level rather than a Settings submenu.

= 1.5.14 =
The postie_post filter has be deprecated in favor of postie_post_before.

= 1.5.3 =
Postie can now set the first image in an email to be the "Featured" image. There is a new setting "Use First Image as Featured Image" which is off by default.
Postie now supports Use Transport Layer Security (TLS)

= 1.5.0 =
New filter postie_filter_email. Used to map "from" to any other email. Allows custom user mapping.

= 1.4.41 =
Post format is now supported. You can specify any of the WordPress supported post formats using the Post type syntax.
Post status can now be specified using the status: tag.
Post status setting was renamed to Default Post Status and moved to the Message tab.

= 1.4.10 =
All script, style and body tags are stripped from html emails.

= 1.4.6 =
Attachments are now processed in the order they were attached.

== CHANGELOG ==
= 1.9.24 (2018-07-12) =
* log functions that use Postie filters
* remove empty tags

= 1.9.23 (2018-05-28) =
* Fix issue when attachment doesn't conform to MIME standard

= 1.9.22 (2018-05-28) =
* Remove Text for Message Start and Text for Message End defaults as they conflicted with some CSS

= 1.9.21 (2018-05-08) =
* Fix: WP is removing backslashes so double them up before saving

= 1.9.21 (2018-04-11) =
* Fix: Compatibility with PHP 5.2-5.4

= 1.9.20 (2018-04-11) =
* Fix: Don't emit attachment div if there is no content after template generation

= 1.9.19 (2018-03-21) =
* Fix: Recognize paragraph line ending in text mode for both styles

= 1.9.18 (2018-03-13) =
* Fix: new signature removal logic removed line endings in some cases

= 1.9.17 (2018-03-12) =
* Fix: Plain text with < > being processed as html causing loss of content during signature removal

= 1.9.16 (2018-02-21) =
* Fix: don't linkify URLs in style attributes

= 1.9.15 (2018-01-25) =
* Fix: accommodate extra spaces in content type header when charset is specified 

= 1.9.14 (2017-12-01) =
* Feature: Additional logging of graphics library in use.

= 1.9.13 (2017-11-22) =
* Feature: New filter postie_include_attachment to allow/prevent attachment to be added to post. See http://postieplugin.com/filter-postie_include_attachment/

= 1.9.12 (2017-11-15) =
* Fix: use wp native function to detect MySQL version

= 1.9.11 (2017-11-10) =
* Fix: remove encoding option, everything is UTF-8.
* Feature: add MySQL version to debug output, suppress many email parsing debug messages
* Fix: don't set actual post format if "standard"
* Feature: Hides WordPress post-by-email feature if Postie is active

= 1.9.10 (2017-10-31) =
* Fix: WP CLI compatibility

= 1.9.9 (2017-10-30) =
* Fix: remove assumption that there is a user with the administrator role

= 1.9.8 (2017-10-23) =
* Fix: check that global is initialized before using

= 1.9.7 (2017-10-19) =
* Fix: more robustness if iconv fails

= 1.9.6 (2017-10-19) =
* Fix: Post format was not being correctly set.

= 1.9.5 (2017-10-13) =
* Feature: change post status to pending/draft if the user doesn't have publish_posts capability
* Fix: process postie_email_notify_recipients filter regardless of the "Send post confirmation email to" setting
* Fix: process postie_email_reject_recipients filter regardless of the "Send post confirmation email to" setting

= 1.9.4 (2017.08.10) =
* Fix: Incorrectly looking for links in html head style blocks

= 1.9.3 (2017.08.04) =
* Feature: Add "every 2 hours" option
* Fix: Include actual response when IMAP authentication fails
* Fix: Incorrectly processing text/html sections as attachments in some cases

= 1.9.2 (2017.07.24) =
* Fix: Uninitialized variable in Postie::create_post()
* Fix: More robust tag detection

= 1.9.1 (2017-06-28) =
* Feature: Add postie_file_added_pre action
* Fix: improvements to tag detection
* Fix: tags may be specified multiple times
* Fix: significant code restructure to bring Postie in line with current coding practices
* Fix: move signature removal to before newline filtering
* Fix: signature removal was deleting all text if the matching regex had whitespace characters

= 1.8.45 (2017-06-21) =
* Feature: If image placement directive has a caption, update the image metadata to include the caption.
* Fix: Only process delay tag if date was found in header
* Fix: Date tag overrides header and no offsets are applied
* Fix: Check for term slug as well as name

= 1.8.44 (2017-06-14) =
* Ensure only users from current site are validated.
* Process newlines earlier in process.
* WordPress 4.8 compatibility testing.

= 1.8.43 (2017-05-10) =
* Fix "postie-attachments" div to wrap the right content
* Fix time zone adjustment

= 1.8.42 (2017-05-10) =
* Wrap all attachments in div with css class "postie-attachments"
* Remove APOP authentication attempt as some servers disconnect on failure.

= 1.8.41 (2017-05-05) =
* Organizational changes in prep for move to more OO approach
* Fix error when time offset is negative.

= 1.8.40 (2017-04-24) =
* Ensure date set to current date when configured
* Compatibility testing with WordPress 4.7.4

= 1.8.39 (2017-04-18) =
* Ensure the time zone is set when calculating the post time.
* Additional debug output

= 1.8.38 (2017-04-06) =
* Update postie_notify_error to be less noisy.

= 1.8.37 (2017-04-05) =
* Remove DateTimeImmutable references to be compatible with old versions of PHP

= 1.8.36 (2017-04-05) =
* Allow user to specify a IMAP folder (supplied by Alejandro Liu)
* Fix postie_file_added action
* New setting "use time offset" which controls whether or not the postie time offset is applied.
* New setting "ignore_email_date" which controls whether or not the email "from" date is used.
* Other email to post date improvements
* New setting "Notify on Error" to control who gets notified on significant errors.
* Email admins on some errors
* Fix bug when parsing headers that have multiple encoded sections

= 1.8.35 (2017-03-21) =
* Check for blank image file extension and replace with mime subtype to get around new filetype checking in WordPress 4.7.2 https://core.trac.wordpress.org/ticket/39550
* Additional logging of message and part headers
* Fix base64 header decoding so trailing space isn't removed
* Fix header decoding to remove extra utf-8 byte order mark
* Add current WordPress time to logging

= 1.8.34 (2017-03-15) =
* Fix bug when sending reject email

= 1.8.33 (2017-03-13) =
* New setting "Remove matched categories" allows category specifiers to remain in post title.
* New filter postie_cleanhtml to determine if Postie should clean up the html or if some other AddOn will do it.
* Specifically call out public methods

= 1.8.32 (2017-03-07) =
* Ensure password not in logs for POP3.
* Additional POP3 header checks.
* Additional POP3 logging.
* Ensure postie_raw always gets called.
* Default to GMT when timezone is blank.
* Use WordPress function to upload images rather than old custom function.
* Remove Generate Thumbnails option as now handled by WordPress.
* ATT00001.txt and its ilk are now always considered a banned attachment regardless of MIME type.
* Removed call to date_default_timezone_set()
* Send notifications to all WP admins if configured.
* Add support for postie_category filter.

= 1.8.31 (2017-02-07) =
* Remove timezone warning if GMT+0
* Add postie_bare_link filter

= 1.8.30 (2017-02-06) =
* Fixed bug where Postie won't run on PHP versions prior to 5.5.

= 1.8.29 (2017-02-06) =
* Fixed bug where "Include Featured Image in Post" = No was not being respected with inline images (iOS mail).

= 1.8.28 (2017-01-26) =
* Fixed bug where having ignored files can cause email body to be blank.

= 1.8.27 (2017-01-26) =
* Added postie_raw action.

= 1.8.26 (2017-01-24) =
* Fix bug where WordPress timezone was not being taken into effect.

= 1.8.25 (2017-01-20) =
* Fix bug where server settings page not compatible with PHP 5.3

= 1.8.24 (2017-01-20) =
* Fix bug where debug output was displaying plaintext passwords.
* Fix bug where categories specified with a colon required a space after the colon.
* Postie now works off blog time rather than GMT.
* Fix bug where oEmbed::get_provider() is not available on WP versions prior to 4.0.

= 1.8.23 (2017-01-12) =
* Fix bug where long subject lines are missing a space.
* Fix bug where {CAPTION} placeholder not being removed if no caption.

= 1.8.22 (2016-12-07) =
* Fix bug where Image Place Holder Tag setting was not respected 

= 1.8.21 (2016-12-07) =
* Clarified the behavior of the "Use custom image field for images" setting and moved it to the Image tab.
* Clarified text on some settings.
* Fix bug which prevented captions from working correctly
* Fix bug which prevented more than 9 #img# references from working.
* Account for unicode non-breaking spaces in regular expressions.

= 1.8.20 (2016-11-21) =
* Fix issue where attachment with Content-ID doesn't have a reference in the html not being added to the post.

= 1.8.19 (2016-11-16) =
* Fix issue where attachments not showing if both plain and html are blank and the attachment is part of a multipart (related) block.

= 1.8.18 (2016-11-16) =
* Fix issue where inline PDF was not being detected as a PDF.
* Fix issue where outlook.com is setting an incorrect mime type on images.

= 1.8.17 (2016-11-03) =
* Fix bugs in cURL connection.

= 1.8.16 (2016-11-03) =
* Fix bug where postie_place_media_before and postie_place_media_after were being called incorrectly.

= 1.8.15 (2016-11-02) =
* Fix bug where inline images were not being replaced correctly if WordPress changed the name of the attachment via sanitize_file_name()

= 1.8.14 (2016-11-01) =
* Fix bug where OS detection failure was preventing email processing

= 1.8.13 (2016-10-31) =
* Fix bug where inline images were not being found due to case differences, e.g. img_4180.jpg vs. IMG_4180.JPG

= 1.8.12 (2016-10-25) =
* Fix bug in postie_email_notify_body filter where the modified body is not used for the email.

= 1.8.11 (2016-10-24) =
* Support Dovecot IMAP server better

= 1.8.10 (2016-10-24) =
* New filter postie_place_media - allows markup to be changed when #img# references are being replaced.
* New filter postie_place_media_before - allows markup to be changed when attachment templates are placed before the content.
* New filter postie_place_media_after - allows markup to be changed when attachment templates are placed after the content.

= 1.8.9 (2016-10-23) =
* Fix bug when calling new postie_email_* filters
* Fix bug where inline and related attachments were not being considered for #img#

= 1.8.8 (2016-10-21) =
* Fix bug where success/fail email notifications had missing info.
* Fix bug where galley wan't being inserted on some plain text messages
* Enhance auto linkification to recognize links like <http://example.com/>
* Separate success and failure emails.
* Fix readme
* New filter: postie_email_reject_recipients - allows recipients list to be altered when sending failure message
* New filter: postie_email_reject_subject - allows the subject to be altered when sending failure message
* New filter: postie_email_reject_body - allows message content to be altered when sending failure message
* New filter: postie_email_notify_recipients - allows recipients list to be altered when sending success message
* New filter: postie_email_notify_subject - allows the subject to be altered when sending success message
* New filter: postie_email_notify_body - allows message content to be altered when sending success message

= 1.8.7 (2016-10-20) =
* Make code compatible with PHP 5.2
* Fix bug where the connection tried to use TLS even though it shouldn't
* Switch to stream API vs sockets since sockets don't seem to be installed typically and the stream API is core.
* Allow self signed certificates with socket connections
* Don't check peer name with socket connections, many hosting companies aren't configured correctly

= 1.8.6 (2016-10-19) =
* Fix bug where the featured image was not included even though config said it should be.
* Ensure Socket connections try to connect with TLS 1.0, 1.1 and 1.2 if SSL is specified.
* Additional logging to help diagnose cURL issue.

= 1.8.5 (2016-10-18) =
* Fix to check if cURL is installed before using

= 1.8.4 (2016-10-17) =
* General release

= 1.8.3 (beta 4) =
* Refactor attachment handling
* Gallery shortcode handling is now correct when there are both images and non-images.

= 1.8.2 (beta 3) =
* New icon set thanks to Chris Lacey
* php-imap replaced by cURL and Socket connection - work sponsored by xsell.net
* TLS automatically detected, setting removed
* Ignore mail state no longer supported, setting removed
* Header array in filter_email3 and postie_post_before filters have changed format
* Fixed paragraph detection in plain text when removing newlines
* Removed old partially functioning forward detection logic
* Transform "[cid:xxx-xx-xx]" references gmail adds to image references.
* Add new filter: postie_post_pre that runs after the email is parsed, but before any changes are made.
* Fixed improper decoding of encoded headers when multiple 'encoded-words' are present.
* Fix to allow multiple #img# reference to the same image.
* The default video 1 template is now 'vshortcode'
* New option: Text fallback. Falls back to plain if html is blank and vice versa.
* Support removing featured image from html

= 1.8.1 (beta 2) =

= 1.8.0 (beta 1) =

= 1.7.32 (2016-04-15) =
* Deal with incorrectly formatted date headers

= 1.7.31 (2016-03-05) =
* Suppress warning message when accessing POP and there are no messages.

= 1.7.30 (2016-02-16) =
* prevent auto-linkifying inside shortcodes

= 1.7.29 (2016-02-15) =
* fixed email header parsing bug in PHP7

= 1.7.28 (2016-02-09) =
* better tag detection with html
* Don't skip image processing when Include Featured Image in Post is set to "No" and the Preferred Text Type is HTML.
* Use the blog name as the "from" text in any emails.
* New filter: postie_filter_email3
* Email headers now available in postie_post_before filter
* When looking for a parent post to add comments ensure the comments are open

= 1.7.27 (2015-12-28) =
* Fix category match settings not saving

= 1.7.26 (2015-12-28) =
* Detect oEmbedable links and don't linkify
* New filter postie_preconnect. http://postieplugin.com/filter-postie_preconnect/

= 1.7.25 (2015-12-15) =
* Fix settings page for new category matching flavors

= 1.7.24 (2015-12-15) =
* Don't process youtube and vimeo links specially.
* New setting to turn off category matching flavors

= 1.7.23 (2015-12-09) =
* Fix bug where emails inside shortcodes were being linkified
* Lookup categories by slug as well as by name
* WordPress 4.4 testing
* Added new video template for using video shortcode
* Added new template variable - FILETYPE which is the file extension

= 1.7.22 (2015-11-09) =
* Update admin screen to match current WP style
* Remove dependence on simple tabs jQuery-UI library and use WP admin tab style
* Remove email password from logging

= 1.7.21 (2015-10-27) =
* Refix bug where "Ignore mail state" setting was being ignored

= 1.7.20 (2015-10-26) =
* Fixed bug where debug info was not being displayed according to settings
* Fix bug where "Ignore mail state" setting was being ignored
* Fix bug where empty post was being generated when already read mail was in the inbox and "Ignore mail state" was "Yes"
* Added postie_session_start and postie_session_end actions

= 1.7.19 (2015-10-13) =
* Fixed bug where allowed mime types was not being respected.

= 1.7.18 (2015-10-13) =
* Fix bug where linkify was messing up CID reference

= 1.7.17 (2015-10-12) =
* New action, postie_log_error
* New action, postie_log_debug
* New feature to turn off all logging
* Only errors logged by default

= 1.7.16 (2015-10-08) =
* Ensure comments are valid html after striping if preferred text type is html

docs/Changes.txt
== Upgrade Notice ==

= 1.8.23 =
Postie now respects the blog timezone, this may require you to change the "Postie Time Correction" setting.

= 1.8.0 =
The php-imap library has been replaced with logic based on Flourish fMailbox et al, there are some differences in the structure of the mail header array. This affects the 
postie_filter_email3 and postie_post_before filters.
See http://flourishlib.com/docs/fMailbox

= 1.6.0 =
Remote cron jobs need to update the URL used to kick off a manual email check. The new URL is http://<mysite>/?postie=get-mail
Accessing http://<mysite>/wp-content/plugins/postie/get_mail.php will now receive a 403 error and a message stating what the new URL should be.
The Postie menu is now at the main level rather than a Settings submenu.

= 1.5.14 =
The postie_post filter has be deprecated in favor of postie_post_before.

= 1.5.3 =
Postie can now set the first image in an email to be the "Featured" image. There is a new setting "Use First Image as Featured Image" which is off by default.
Postie now supports Use Transport Layer Security (TLS)

= 1.5.0 =
New filter postie_filter_email. Used to map "from" to any other email. Allows custom user mapping.

= 1.4.41 =
Post format is now supported. You can specify any of the WordPress supported post formats using the Post type syntax.
Post status can now be specified using the status: tag.
Post status setting was renamed to Default Post Status and moved to the Message tab.

= 1.4.10 =
All script, style and body tags are stripped from html emails.

= 1.4.6 =
Attachments are now processed in the order they were attached.

== CHANGELOG ==
= 1.9.24 (2018-07-12) =
* log functions that use Postie filters
* remove empty tags

= 1.9.23 (2018-05-28) =
* Fix issue when attachment doesn't conform to MIME standard

= 1.9.22 (2018-05-28) =
* Remove Text for Message Start and Text for Message End defaults as they conflicted with some CSS

= 1.9.21 (2018-05-08) =
* Fix: WP is removing backslashes so double them up before saving

= 1.9.21 (2018-04-11) =
* Fix: Compatibility with PHP 5.2-5.4

= 1.9.20 (2018-04-11) =
* Fix: Don't emit attachment div if there is no content after template generation

= 1.9.19 (2018-03-21) =
* Fix: Recognize paragraph line ending in text mode for both styles

= 1.9.18 (2018-03-13) =
* Fix: new signature removal logic removed line endings in some cases

= 1.9.17 (2018-03-12) =
* Fix: Plain text with < > being processed as html causing loss of content during signature removal

= 1.9.16 (2018-02-21) =
* Fix: don't linkify URLs in style attributes

= 1.9.15 (2018-01-25) =
* Fix: accommodate extra spaces in content type header when charset is specified 

= 1.9.14 (2017-12-01) =
* Feature: Additional logging of graphics library in use.

= 1.9.13 (2017-11-22) =
* Feature: New filter postie_include_attachment to allow/prevent attachment to be added to post. See http://postieplugin.com/filter-postie_include_attachment/

= 1.9.12 (2017-11-15) =
* Fix: use wp native function to detect MySQL version

= 1.9.11 (2017-11-10) =
* Fix: remove encoding option, everything is UTF-8.
* Feature: add MySQL version to debug output, suppress many email parsing debug messages
* Fix: don't set actual post format if "standard"
* Feature: Hides WordPress post-by-email feature if Postie is active

= 1.9.10 (2017-10-31) =
* Fix: WP CLI compatibility

= 1.9.9 (2017-10-30) =
* Fix: remove assumption that there is a user with the administrator role

= 1.9.8 (2017-10-23) =
* Fix: check that global is initialized before using

= 1.9.7 (2017-10-19) =
* Fix: more robustness if iconv fails

= 1.9.6 (2017-10-19) =
* Fix: Post format was not being correctly set.

= 1.9.5 (2017-10-13) =
* Feature: change post status to pending/draft if the user doesn't have publish_posts capability
* Fix: process postie_email_notify_recipients filter regardless of the "Send post confirmation email to" setting
* Fix: process postie_email_reject_recipients filter regardless of the "Send post confirmation email to" setting

= 1.9.4 (2017.08.10) =
* Fix: Incorrectly looking for links in html head style blocks

= 1.9.3 (2017.08.04) =
* Feature: Add "every 2 hours" option
* Fix: Include actual response when IMAP authentication fails
* Fix: Incorrectly processing text/html sections as attachments in some cases

= 1.9.2 (2017.07.24) =
* Fix: Uninitialized variable in Postie::create_post()
* Fix: More robust tag detection

= 1.9.1 (2017-06-28) =
* Feature: Add postie_file_added_pre action
* Fix: improvements to tag detection
* Fix: tags may be specified multiple times
* Fix: significant code restructure to bring Postie in line with current coding practices
* Fix: move signature removal to before newline filtering
* Fix: signature removal was deleting all text if the matching regex had whitespace characters

= 1.8.45 (2017-06-21) =
* Feature: If image placement directive has a caption, update the image metadata to include the caption.
* Fix: Only process delay tag if date was found in header
* Fix: Date tag overrides header and no offsets are applied
* Fix: Check for term slug as well as name

= 1.8.44 (2017-06-14) =
* Ensure only users from current site are validated.
* Process newlines earlier in process.
* WordPress 4.8 compatibility testing.

= 1.8.43 (2017-05-10) =
* Fix "postie-attachments" div to wrap the right content
* Fix time zone adjustment

= 1.8.42 (2017-05-10) =
* Wrap all attachments in div with css class "postie-attachments"
* Remove APOP authentication attempt as some servers disconnect on failure.

= 1.8.41 (2017-05-05) =
* Organizational changes in prep for move to more OO approach
* Fix error when time offset is negative.

= 1.8.40 (2017-04-24) =
* Ensure date set to current date when configured
* Compatibility testing with WordPress 4.7.4

= 1.8.39 (2017-04-18) =
* Ensure the time zone is set when calculating the post time.
* Additional debug output

= 1.8.38 (2017-04-06) =
* Update postie_notify_error to be less noisy.

= 1.8.37 (2017-04-05) =

config_form_server.php
<div id="simpleTabs-content-1" class="simpleTabs-content">

    <table class='form-table'>
        <tr>
            <th scope="row"><lable for="postie-settings-input_connection"><?php _e('Connection', 'postie') ?></lable></th>
        <td>
            <select name='postie-settings[input_connection]' id='postie-settings-input_connection'>
                <option value="sockets"  <?php echo (($input_connection == "socket") ? " selected='selected' " : "") ?>>Sockets</option>
                <?php if (function_exists('curl_version')) { ?>
                    <option value="curl" <?php echo ($input_connection == "curl") ? "selected='selected' " : "" ?>>cURL</option>
                <?php } ?>
            </select>
            <p class='description'><?php _e("Sockets is preferred, but doesn't work with some hosts.", 'postie'); ?></p>
        </td>
        </tr>

        <tr>
            <th scope="row"><lable for="postie-settings-input_protocol"><?php _e('Mail Protocol', 'postie') ?></lable></th>
        <td>
            <select name='postie-settings[input_protocol]' id='postie-settings-input_protocol'>
                <option value="pop3"  <?php echo (($input_protocol == "pop3") ? " selected='selected' " : "") ?>>POP3</option>
                <option value="imap" <?php echo ($input_protocol == "imap") ? "selected='selected' " : "" ?>>IMAP</option>
                <option value="pop3-ssl" <?php echo ($input_protocol == "pop3-ssl") ? "selected='selected' " : "" ?>>POP3-SSL</option>
                <option value="imap-ssl" <?php echo ($input_protocol == "imap-ssl") ? "selected='selected' " : "" ?>>IMAP-SSL</option>
            </select>

        </td>
        </tr>

        <tr>
            <th scope="row"><label for="postie-settings-mail_server_port"><?php _e('Port', 'postie') ?></label></th>
            <td valign="top">
                <input name='postie-settings[mail_server_port]' style="width: 70px;" type="number" min="0" id='postie-settings-mail_server_port' value="<?php echo esc_attr($mail_server_port); ?>" size="6" />
                <p class='description'><?php _e("Standard Ports:", 'postie'); ?><br />
                    <?php _e("POP3", 'postie'); ?>: 110<br />
                    <?php _e("IMAP", 'postie'); ?>: 143<br />
                    <?php _e("IMAP-SSL", 'postie'); ?>: 993 <br />
                    <?php _e("POP3-SSL", 'postie'); ?>: 995 <br />
                </p>
            </td>
        </tr>
        <tr valign="top">
            <th scope="row"><?php _e('Mail Server', 'postie') ?></th>
            <td><input name='postie-settings[mail_server]' type="text" id='postie-settings-mail_server' value="<?php echo esc_attr($mail_server); ?>" size="40" />
            </td>
        </tr>
        <tr valign="top">
            <th scope="row"><?php _e('Mail Userid', 'postie') ?></th>
            <td><input name='postie-settings[mail_userid]' type="text" id='postie-settings-mail_userid' autocomplete='new-password' value="<?php echo esc_attr($mail_userid); ?>" size="40" /></td>
        </tr>
        <tr valign="top">
            <th scope="row"><?php _e('Mail Password', 'postie') ?></th>
            <td>
                <input name='postie-settings[mail_password]' type="password" id='postie-settings-mail_password' autocomplete='new-password' value="<?php echo esc_attr($mail_password); ?>" size="40" />
            </td>
        </tr>

        <?php echo PostieAdmin::boolean_select_html(__("Ignore Email Date", 'postie'), 'postie-settings[ignore_email_date]', $ignore_email_date, __("If set to 'Yes' the email date will be ignored and the post will be published with the system time. If set to 'No' the date in the email will be used as the post date.", 'postie')); ?>
        <?php echo PostieAdmin::boolean_select_html(__("Use Postie Time Correction", 'postie'), 'postie-settings[use_time_offset]', $use_time_offset, __("If set to 'Yes' adjust the time according to Postie Time Correction otherwise use the date provided by the email.", 'postie')); ?>

        <tr>
            <th scope="row"><?php _e('Postie Time Correction', 'postie') ?></th>
            <td><input style="width: 70px;" name='postie-settings[time_offset]' type="number" step="0.5" id='postie-settings-time_offset' size="2" value="<?php echo esc_attr($time_offset); ?>" /> 
                <?php
                _e('hours', 'postie');
                ?> 
                <p class='description'><?php printf(__("Should be the same as your normal offset, but this lets you adjust it in cases where that doesn't work.<br>Blog timezone is: %s<br>Blog offset: %d", 'postie'), get_option('timezone_string') == '' ? 'GMT+0' : get_option('timezone_string'), get_option('gmt_offset')); ?></p>
            </td>
        </tr>
        <tr>
            <th>
                <?php _e('Check for mail every', 'postie') ?>:
            </th>
            <td>
                <select name='postie-settings[interval]' id='postie-settings-interval'>
                    <option value="weekly" <?php
                    if ($interval == "weekly") {
                        echo "selected='selected'";
                    }
                    ?>><?php _e('Once weekly', 'postie') ?>
                    </option>

                    <option value="daily"<?php
                    if ($interval == "daily") {
                        echo "selected='selected'";
                    }
                    ?>><?php _e('daily', 'postie') ?>
                    </option>
                    
                      <option value="twohours"<?php
                    if ($interval == "twohours") {
                        echo "selected='selected'";
                    }
                    ?>><?php _e('every 2 hours', 'postie') ?>
                    </option>

                    <option value="hourly" <?php
                    if ($interval == "hourly") {
                        echo "selected='selected'";
                    }
                    ?>><?php _e('hourly', 'postie') ?>
                    </option>

                    <option value="twiceperhour" <?php
                    if ($interval == "twiceperhour") {
                        echo "selected='selected'";
                    }
                    ?>><?php _e('every 30 minutes', 'postie') ?>
                    </option>

                    <option value="tenminutes" <?php
                    if ($interval == "tenminutes") {
                        echo "selected='selected'";
                    }
                    ?>><?php _e('every 10 minutes', 'postie') ?>
                    </option>

                    <option value="fiveminutes" <?php
                    if ($interval == "fiveminutes") {
                        echo "selected='selected'";
                    }
                    ?>><?php _e('every 5 minutes', 'postie') ?>
                    </option>

                    <option value="oneminute" <?php
                    if ($interval == "oneminute") {
                        echo "selected='selected'";
                    }
                    ?>><?php _e('every 1 minute', 'postie') ?>
                    </option>

                    <option value="thirtyseconds" <?php
                    if ($interval == "thirtyseconds") {
                        echo "selected='selected'";
                    }
                    ?>><?php _e('every 30 seconds', 'postie') ?>
                    </option>

                    <option value="fifteenseconds" <?php
                    if ($interval == "fifteenseconds") {
                        echo "selected='selected'";
                    }
                    ?>><?php _e('every 15 seconds', 'postie') ?>
                    </option>

                    <option value="manual" <?php
                    if ($interval == "manual") {
                        echo "selected='selected'";
                    }
                    ?>><?php _e('check manually', 'postie') ?>
                    </option>
                </select>
            </td>
        </tr>
        <tr>
            <th>
                <?php _e('Maximum number of emails to process', 'postie'); ?>
            </th>
            <td>
                <select name='postie-settings[maxemails]' id='postie-settings-maxemails'>
                    <option value="0" <?php if ($maxemails == '0') echo "selected='selected'"; ?>><?php _e('All', 'postie'); ?></option>
                    <option value="1" <?php if ($maxemails == '1') echo "selected='selected'" ?>>1</option>
                    <option value="2" <?php if ($maxemails == '2') echo "selected='selected'" ?>>2</option>
                    <option value="5" <?php if ($maxemails == '5') echo "selected='selected'" ?>>5</option>
                    <option value="10" <?php if ($maxemails == '10') echo "selected='selected'" ?>>10</option>
                    <option value="25" <?php if ($maxemails == '25') echo "selected='selected'" ?>>25</option>
                    <option value="50" <?php if ($maxemails == '50') echo "selected='selected'" ?>>50</option>
                </select>
            </td>
        </tr>
        <?php
        echo PostieAdmin::boolean_select_html(__("Delete email after posting", 'postie'), 'postie-settings[delete_mail_after_processing]', $delete_mail_after_processing, __("Only set to no for testing purposes", 'postie'));
        //echo PostieAdmin::BuildBooleanSelect(__("Ignore mail state", 'postie'), 'postie-settings[ignore_mail_state]', $ignore_mail_state, __("Ignore whether the mails is 'read' or 'unread' If 'No' then only unread messages are processed. IMAP only", 'postie')); 

        echo PostieAdmin::boolean_select_html(__("Enable Error Logging", 'postie'), 'postie-settings[postie_log_error]', $postie_log_error, __("Log error messages to the web server error log.", 'postie'));

        $notification_options = array('(Nobody)', '(All Admins)');
        foreach (get_users(array('role' => 'administrator')) as $user) {
            $notification_options[] = $user->user_login;
        }
        echo PostieAdmin::select_html(__('Notify on Error', 'postie'), 'postie-settings[postie_log_error_notify]', $postie_log_error_notify, $notification_options);

        echo PostieAdmin::boolean_select_html(__("Enable Debug Logging", 'postie'), 'postie-settings[postie_log_debug]', $postie_log_debug, __("Log debug messages to the web server error log.", 'postie'));
        ?>

    </table>
</div>